### PR TITLE
many: refactor registry errors and add GetView function

### PR DESCRIPTION
**Pull request by MiguelPires**
Refactor the not found errors as we now also have to use them in registrystate so the previous construction is not convenient. Also improves clarity and consistency in some of the errors messages. The second commit adds a GetView function to registrystate so it can be used in several places and removes some usages of Set from ctlcmd tests (so we can remove it in the next PR when we stop using it in the API). This became a bit of a ball of mud of refactorings but I'd like to do them before moving on to the next epics

### CODE REVIEW

**Review by andrewphelpsj — APPROVED** [review]
Thanks!

File: overlord/hookstate/ctlcmd/export_test.go [hunk 1]
```diff
@@ -188,10 +188,26 @@ func MockRegistrystateGetTransaction(f func(*registrystate.Context, *state.State
 	}
 }
 
-func MockGetRegistryView(f func(ctx *hookstate.Context, account, registryName, viewName string) (*registry.View, error)) (restore func()) {
-	old := getRegistryView
-	getRegistryView = f
+func MockRegistrystateGetView(f func(st *state.State, account, registryName, viewName string) (*registry.View, error)) (restore func()) {
+	old := registrystateGetView
```
Comment (ZeyadYasser) [inline overlord/hookstate/ctlcmd/export_test.go:192]: tiny nitpick, `testutil.Mock` would simplify this and the ones added below

**Review by ZeyadYasser — APPROVED** [review]
LGTM, Thank you!

**Review by olivercalder — APPROVED** [review]
Looks very nice, thanks!

**Comment by MiguelPires** [comment]
Failed tests:
* Prepare fedora-40-64:tests/main/
* Task ubuntu-core-16-64:tests/main/core-snap-refresh-with-shared-memory
* Prepare ubuntu-18.04-64:tests/main/apparmor-prompting-integration-tests:write_single_deny
* Prepare ubuntu-20.04-64:tests/main/apparmor-prompting-integration-tests
* Prepare ubuntu-core-20-64:tests/main/apparmor-prompting-integration-tests
* Prepare ubuntu-core-24-64:tests/main/apparmor-prompting-integration-tests:create_multiple_actioned_by_other_pid_always_allow
* Prepare ubuntu-24.04-64:tests/main/snapd-update-services
* Prepare amazon-linux-2-64:tests/main/classic-custom-device-reg

**Pull request merged**